inc/Ui/Recent.php

```diff
@@ -162,7 +162,7 @@ protected function checkCurrentRevision(array &$info)
         }
         if (!$changelog->isCurrentRevision($info['date'])) {
             $currentRevInfo = $changelog->getCurrentRevisionInfo();
-            if ($currentRevInfo['type'] == DOKU_CHANGE_TYPE_DELETE) {
+            if (($currentRevInfo['type'] ?? null) == DOKU_CHANGE_TYPE_DELETE) {
```

Comment:
If I understand correctly, `getCurrentRevisionInfo()` may return `null`, but it's not like it will return an array that is missing the `type` key. So I think the code would be clearer as
```
if ($currentRevInfo && $currentRevInfo['type'] == DOKU_CHANGE_TYPE_DELETE) {
```

Reply:
No, in this case `getCurrentRevisionInfo()` returned a `bool`(`false`) not `null`. It is documented that it may do so. (But it may not return `null`.)

But your version of the fix would work if it is guaranteed that if it returns an `array` that said `array` contains the key `'type'`. According to its documentation that is the case. I.e. returning an `array` without the key `'type'` would violate the API contract.

So while I think my fix guards against more failure scenarios, strictly speaking yours is correct.